### PR TITLE
 Anchors: string start ^ and end $

### PR DESCRIPTION
**Pull request by EhsanShahbazii**
/done

### CODE REVIEW

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 1]
```diff
@@ -1,52 +1,51 @@
...
 ```js run
 let str1 = "its fleece was white as snow";
-alert( /snow$/.test(str1) ); // true
+alert( /snow$/.test(str1) ); // درست
 ```
 
-In these particular cases we could use string methods `startsWith/endsWith` instead. Regular expressions should be used for more complex tests.
-
-## Testing for a full match
+در این موارد خاص، می‌توانیم به جای آن از روش‌های رشته‌ای `startsWith/endsWith` استفاده کنیم. برای تست های پیچیده تر باید از عبارات منظم استفاده شود.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:23]:
```suggestion
در این موارد خاص، می‌توانیم به جای آن از متدهای مختص به رشته‌ها به نام `startsWith/endsWith` استفاده کنیم. برای تست های پیچیده تر باید از عبارات منظم استفاده شود.
```
Try to improve the readability of the translation even if you need to insert additional words to it.

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 2]
```diff
@@ -1,52 +1,51 @@
...
 
-Both anchors together `pattern:^...$` are often used to test whether or not a string fully matches the pattern. For instance, to check if the user input is in the right format.
+## تست برای یک مسابقه کامل
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:25]:
```suggestion
## آزمایش یک انطباق کامل
```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 3]
```diff
@@ -1,52 +1,51 @@
...
 
-Let's check whether or not a string is a time in `12:34` format. That is: two digits, then a colon, and then another two digits.
+هر دو لنگر با هم `$...^:pattern` اغلب برای آزمایش اینکه آیا یک رشته کاملاً با الگو مطابقت دارد یا نه استفاده می شود. به عنوان مثال، برای بررسی اینکه آیا ورودی کاربر در قالب مناسب است یا خیر.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:27]:
```suggestion
هر دو anchor یعنی `$...^:pattern` با هم اغلب برای آزمایش اینکه آیا یک رشته کاملاً با الگو مطابقت دارد یا نه استفاده می شود. به عنوان مثال، برای بررسی اینکه آیا ورودی کاربر در قالب مناسب است یا خیر.
```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 4]
```diff
@@ -1,52 +1,51 @@
...
 
-In regular expressions language that's `pattern:\d\d:\d\d`:
+بیایید بررسی کنیم که آیا یک رشته زمان در قالب `12:34` است یا خیر. یعنی: دو رقم، سپس یک دونقطه، و سپس دو رقم دیگر.
 
+در زبان عبارات معمولی `pattern:\d\d:\d\d` است:
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:31]:
```suggestion
در زبان عبارات با قاعده به صورت `pattern:\d\d:\d\d` است:

```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 5]
```diff
@@ -1,52 +1,51 @@
...
 ```js run
 let goodInput = "12:34";
 let badInput = "12:345";
 
 let regexp = /^\d\d:\d\d$/;
-alert( regexp.test(goodInput) ); // true
-alert( regexp.test(badInput) ); // false
+alert( regexp.test(goodInput) ); // درست
+alert( regexp.test(badInput) ); // نادرست
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:38]:
```suggestion
alert( regexp.test(goodInput) ); // true
alert( regexp.test(badInput) ); // false
```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 6]
```diff
@@ -1,52 +1,51 @@
...
 ```
 
-Here the match for `pattern:\d\d:\d\d` must start exactly after the beginning of the text `pattern:^`, and the end `pattern:$` must immediately follow.
+در اینجا تطبیق `pattern:\d\d:\d\d` باید دقیقاً بعد از ابتدای متن `^:pattern` شروع شود و انتهای `$:pattern` باید بلافاصله دنبال شود.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:41]:
```suggestion
در اینجا تطبیق `pattern:\d\d:\d\d` باید دقیقاً بعد از ابتدای متن `^:pattern` شروع شود و انتهای `$:pattern` باید بلافاصله به دنبال آن بیاید.
```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 7]
```diff
@@ -1,52 +1,51 @@
...
 
-The whole string must be exactly in this format. If there's any deviation or an extra character, the result is `false`.
+کل رشته باید دقیقاً در این قالب باشد. اگر انحراف یا یک کاراکتر اضافی وجود داشته باشد، نتیجه `نادرست` است.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:43]:
```suggestion
کل رشته باید دقیقاً در این قالب باشد. اگر انحراف یا یک کاراکتر اضافی وجود داشته باشد، نتیجه `false` است.
```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 8]
```diff
@@ -1,52 +1,51 @@
...
 
-Anchors behave differently if flag `pattern:m` is present. We'll see that in the next article.
+اگر پرچم `pattern:m` وجود داشته باشد، لنگرها رفتار متفاوتی دارند. آن را در مقاله بعدی خواهیم دید.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:45]:
```suggestion
اگر پرچم `pattern:m` وجود داشته باشد، anchorها رفتار متفاوتی دارند. آن را در مقاله بعدی خواهیم دید.
```

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 9]
```diff
@@ -1,52 +1,51 @@
...
 
 ```smart header="Anchors have \"zero width\""
-Anchors `pattern:^` and `pattern:$` are tests. They have zero width.
+لنگرهای `^:pattern` و `$:pattern` تست هستند. عرض آنها صفر است.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/article.md:48]:
```suggestion
anchorهای `^:pattern` و `$:pattern` تست هستند. عرض آنها صفر است.
```

File: 9-regular-expressions/04-regexp-anchors/1-start-end/solution.md [hunk 10]
```diff
@@ -1,5 +1,5 @@
-An empty string is the only match: it starts and immediately finishes.
+یک رشته خالی تنها منطبق است: شروع می شود و بلافاصله تمام می شود.
 
-The task once again demonstrates that anchors are not characters, but tests.
+این کار یک بار دیگر نشان می دهد که مجریان کاراکترها نیستند، بلکه آزمایش هستند.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/1-start-end/solution.md:3]:
```suggestion
این کار یک بار دیگر نشان می دهد که anchorها کاراکتر نیستند، بلکه آزمایش هستند.
```

File: 9-regular-expressions/04-regexp-anchors/1-start-end/solution.md [hunk 11]
```diff
@@ -1,5 +1,5 @@
...
 
-The string is empty `""`. The engine first matches the `pattern:^` (input start), yes it's there, and then immediately the end `pattern:$`, it's here too. So there's a match.
+رشته `""` خالی است. موتور ابتدا با `^:pattern` (شروع ورودی) مطابقت دارد، بله وجود دارد، و بلافاصله با `$:pattern` انتهایی، آن هم اینجاست. بنابراین یک مسابقه وجود دارد.
```
Comment (mahdiHash) [inline 9-regular-expressions/04-regexp-anchors/1-start-end/solution.md:5]:
```suggestion
رشته `""` خالی است. موتور ابتدا `^:pattern` (شروع ورودی) را تطبیق می‌دهد، بله تطابق دارد، و بلافاصله `$:pattern` انتهایی را، آن هم مطابقت دارد. بنابراین یک انطباق داریم.
```

**Review by mahdiHash — CHANGES_REQUESTED** [review]
Please review the changes.
There must be an empty line after line 8. Please pay a good attention to line syncing, I may have missed one or two.
You can add the suggestions to a batch and submit a single commit ([docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes)).

**Comment by mahdiHash** [comment]
As far as I've seen till now, the words "reference" and "match" can be the candidates of "being a tough one to translate" contest!

File: 9-regular-expressions/04-regexp-anchors/article.md [hunk 12]
```diff
@@ -1,52 +1,51 @@
-# Anchors: string start ^ and end $
+# لنگرها: رشته شروع ^ و پایان $
```
[the remark this answers is not in the document]
Comment (mahdyar) [reply 9-regular-expressions/04-regexp-anchors/article.md:1]: > I think "anchor" is better not to be translated. @mahdyar, what's your opinion?

I agree.

**Comment by mahdiHash** [comment]
Alright @EhsanShahbazii, please make the requested changes and comment "/done" when you are.

**Review by mahdiHash — APPROVED** [review]

**Comment by mahdiHash** [comment]
Thanks for your contribution, @EhsanShahbazii.

**Review by mahdyar — APPROVED** [review]

**Pull request merged**